naga/src/proc/constant_evaluator.rs

```diff
@@ -2198,6 +2227,22 @@ impl<'a> ConstantEvaluator<'a> {
             crate::valid::check_literal_value(literal)?;
         }
 
+        // Ensure vector composes contain the correct number of components. We
```

Comment:
Maybe update the comment just above saying "It suffices to only check literals".

Reply:
yeah this is a bit confusing. The comment is still applicable, it's just referring specifically to the `check_literal_value()` check above, rather than all checks in this function (of which there are now multiple). I'll make it clear.